sdk/core/core/src/az_http_private.h

```diff
@@ -43,6 +43,25 @@ AZ_NODISCARD AZ_INLINE az_result _az_http_request_remove_retry_headers(_az_http_
   return AZ_OK;
 }
 
+/**
+ * @brief sets buffer and parser to its initial state
+ *
+ */
+AZ_NODISCARD AZ_INLINE az_result _az_http_response_reset(az_http_response* http_response)
+{
+  if (az_span_length(http_response->_internal.http_response) > 0)
+  {
+    // init response with same buffer and make internal parser to be initial state
+    AZ_RETURN_IF_FAILED(az_http_response_init(
```

Comment:
Just call *_init and ignore the return.

Reply:
You have to check the return, the function is NO_DISCARD.

Reply:
Can we just (void) the return?
Reset should never fail and should therefore return void.

Reply:
I'd rather not, Ignoring the return code makes the code brittle to future enhances to az_http_response_init
I agree az_http_response_init can't fail as its written now.  Should we change that logic in the future I'd expect this function to fail accordingly.